Gems/PythonAssetBuilder/Editor/Scripts/scene_api/common_rules.py

```diff
@@ -0,0 +1,212 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+import json, uuid
+
+class BaseRule():
+    """
+    Base class of the actor rules to help encode the type name of abstract rules
+
+    Parameters
+    ----------
+    typename : str
+        A typename the $type will be in the JSON chunk object
+
+    Attributes
+    ----------
+    typename: str
+        The type name of the abstract classes to be serialized
+
+    id: UUID
+        a unique ID for the rule
+
+    Methods
+    -------
+    to_dict()
+        Converts contents to a Python dictionary
+        Adds the '$type' member
+        Adds a random 'id' member
+        Note: Override this method if a derviced class needs to return a custom dictionary
+        
+    """
+    def __init__(self, typename):
+        self.typename = typename
+        self.id = uuid.uuid4()
+        
+    def __eq__(self, other):
+        return self.id.__eq__(other.id)
+
+    def __ne__(self, other):
+        return self.__eq__(other) is False
+
+    def __hash__(self):
+        return self.id.__hash__()
+    
+    def to_dict(self):
+        data = vars(self)
+        data['id'] = f"{{{str(self.id)}}}"
+        # rename 'typename' to '$type'
+        data['$type'] = self.typename
+        data.pop('typename')
+        return data
+    
+def convert_rule__to_json(rule:BaseRule, indentValue=0):
```

Comment:
there are two underscores in a row after rule

Reply:
yep, will fix